File: src/main/java/io/bindernews/thegrackle/power/PlaguePower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.thegrackle.GrackleMod;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class PlaguePower extends BasePower {
    public static final String POWER_ID = GrackleMod.makeId(PlaguePower.class);

    public PlaguePower(AbstractCreature owner, int amount) {
        super(POWER_ID);
        setOwnerAmount(owner, amount);
        loadRegion("ai");
        updateDescription();
    }

    @Override
    public void atStartOfTurn() {
        addToBot(iop().actionApplyPower(owner, owner, MultiHitPower.POWER_ID, amount));
    }

    @Override
    public void updateDescription() {
        description = formatDesc(0, amount);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/power/PlaguePower.kt
package io.bindernews.thegrackle.power

import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.thegrackle.GrackleMod
import io.bindernews.thegrackle.helper.ModInterop.Companion.iop

class PlaguePower(owner: AbstractCreature, amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, amount)
        loadRegion("ai")
        updateDescription()
    }

    override fun atStartOfTurn() {
        addToBot(iop().actionApplyPower(owner, owner, MultiHitPower.POWER_ID, amount))
    }

    override fun updateDescription() {
        description = formatDesc(0, amount)
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(PlaguePower::class.java)
    }
}